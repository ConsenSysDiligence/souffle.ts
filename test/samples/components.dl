// d.R,d.c.R1
.comp A {
  .decl R(x: number)
  R(1).
  R(2).

  .comp B {
    .decl R1(x: number)

    R1(x) :- R(y), x=y+1.

    .output R1()
  }

  .init c = B
  .output R()
}

.init d = A
