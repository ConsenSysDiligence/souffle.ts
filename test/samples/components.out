/// d.R
===============
x
1
2
/// d.c.R1
===============
x
2
3